### DISCUSSION TITLE
Add `.` as an alias from the current commit

### DISCUSSION THREAD

**Discussion by banool** [arxanas/git-branchless#457] [Ideas]
Let's say I'm looking at a source control tree like this:
```
┃ ◯ da3905b 1h [API] Make a BCS payload type todo undo cargo stuff when new version of poem released to crates.io
┃ ┣━┓
┃ ┃ ✕ 3efc5b0 51m (rewritten as de895cbc) [API] Add request and response types for the JSON / BCS enum
┃ ┃ ┃
┃ ┃ ◯ 5bb408b 25m [API] Add headers to response in Poem backend
┃ ┃ ┃
┃ ┃ ◯ 0cac333 24m (banool/generate_openapi) [API] Add API for accounts for Poem backend
┃ ┃
┃ ● de895cb 53s [API] Add request and response types for the JSON / BCS enum
```

I'm currently on `de895cb`. If I want to rebase `5bb408b` onto my current commit, I have to do this:
```
git move -s 5bb408b -d de895cb
```

Instead I'd love to be able to do this:
```
git move -s 5bb408b -d .
```

This is a pretty handy feature in other VCS systems like Mercurial.

**Comment by arxanas** [comment]
For the specific case of updating dependent commits, you can run [`git restack`](https://github.com/arxanas/git-branchless/wiki/Command:-git-restack) to resolve the situation in your example. https://github.com/arxanas/git-branchless/issues/2 has an item that suggests running `git restack` when the smartlog shows commits that could be restacked, which is maybe worth prioritizing.

For `git move` in general, [the default arguments are `--base HEAD` and `--dest HEAD`](https://github.com/arxanas/git-branchless/wiki/Command:-git-move#default-options), so in this case, you can also write just `git move -s 5bb408b` to mean `git move -s 5bb408b -d .`. It might be a good idea to surface a hint when the user supplies `HEAD` for either `--base` or `--dest`.

In the latest master, git-branchless supports revsets ([brief reference here](https://github.com/arxanas/git-branchless/wiki/Reference:-Revsets)), for which both `@` and `.` are aliases for `HEAD`, so you'll be able to run `git move -s 5bb408b -d .` directly. If you're interested in trying it out, you can [install from source](https://github.com/arxanas/git-branchless/wiki/Runbook#manual-testing). If you do, I'd recommend you try out @claytonrcarter's branch at https://github.com/arxanas/git-branchless/pull/450, which adds a very flexible `-x`/`--exact` option to do more granular commit operations.

**Reply by banool** [reply]
Hey @arxanas, thanks for sharing the info about defaulting to HEAD, I didn't know that. Glad to see that we have `.` as an alias for HEAD soon, that'll be a nice little usability win. Thanks!
